api/api.go

```diff
@@ -166,6 +170,17 @@ func New(
 		if err != nil {
 			return nil, errSysCerts
 		}
+		if opts.CertFile != "" {
+			revProxyCert, err := os.ReadFile(opts.CertFile)
+			if err != nil {
+				c.doLog(log.WithError(err).Error, "Unable to read certificate file")
+				return nil, err
+			}
+			if ok := pool.AppendCertsFromPEM(revProxyCert); !ok {
```

Comment:
Should we consider decoding and validating the certificate before attempting to append it to the pool? What are your thoughts on this approach?

Reply:
AppendCertsFromPEM attempts to parse a series of PEM encoded certificates. It appends any certificates found to s and reports whether any certificates were successfully parsed.

Reply:
How about an expired certificate? Will we get any error if we are trying to add an expired one to the pool?

Reply:
To conclude this discussion, the method will allow both valid and expired certificates to be added to the pool. However, during the TLS handshake, the `x509` package will handle the expiration check and ensure that expired certificates do not affect the connection. Therefore, we can skip the certificate validation at this stage. 

That said, it is always a good practice to perform such checks beforehand to ensure only valid certificates are added. Nonetheless, there is also the possibility that a valid certificate added to the pool today could expire later, and since the certificate pool is cached, it won't refresh until the system is restarted.